src/main/java/duke/logic/Storage.java

```diff
@@ -0,0 +1,120 @@
+     */
+    public Storage(String filePath) throws DukeException {
+        this.file = new File(filePath);
+
+        try {
+            if (!this.file.exists()) {
+                this.file.getParentFile().mkdirs();
+                this.file.createNewFile();
+            }
+        } catch (IOException e) {
+            throw new DukeException("INVALID FILE PATH");
+        }
+    }
+
+    /**
+     * Clears the local file associated with the storage.
+     *
+     * @throws DukeException If write to local file is unsuccessful.
+     */
+    public void clearFile() throws DukeException {
+        this.writeToFile(new TaskList());
+    }
+
+    /**
+     * Writes to the local file associated with the storage.
+     *
+     * @param taskList List of tasks that is to be written to local file.
+     * @throws DukeException If write to local file is unsuccessful.
+     */
+    public void writeToFile(TaskList taskList) throws DukeException {
+        try {
+            FileWriter fw = new FileWriter(this.file);
+            fw.write(taskList.toData());
+            fw.close();
+        } catch (IOException e) {
+            throw new DukeException("INVALID FILE PATH");
+        }
+    }
+
+    /**
+     * Reads from the local file associated with the storage.
+     *
+     * @return List of tasks read from local file.
+     * @throws DukeException If read from local file is unsuccessful.
+     */
+    public ArrayList<Task> readFromFile() throws DukeException {
+        ArrayList<Task> tasks = new ArrayList<>();
+
+        try {
+            Scanner sc = new Scanner(this.file);
+
+            while (sc.hasNextLine()) {
+                String line = sc.nextLine();
+                String[] breakdown = line.split(" \\| ");
+                String command = breakdown[0];
+
+                try {
+                    TaskType taskType = TaskType.valueOf(command);
+
+                    switch (taskType) {
+                    case T:
+                        tasks.add(new Todo(breakdown[2],
+                                Boolean.parseBoolean(breakdown[1])));
+                        break;
+                    case E:
+                        tasks.add(new Event(breakdown[2],
+                                Boolean.parseBoolean(breakdown[1]),
+                                LocalDateTime.parse(breakdown[3])));
+                        break;
+                    case D:
+                        tasks.add(new Deadline(breakdown[2],
+                                Boolean.parseBoolean(breakdown[1]),
+                                LocalDateTime.parse(breakdown[3])));
+                        break;
+                    }
+                } catch (IllegalArgumentException e) {
+                    throw new DukeException("INVALID TASK TYPE");
+                }
+            }
+
+            return tasks;
+        } catch (IOException e) {
+            throw new DukeException("INVALID FILE PATH");
+        }
```

Comment:
```suggestion
        try {
            Scanner sc = new Scanner(this.file);
        } catch (IOException e) {
            throw new DukeException("INVALID FILE PATH");
        }

        while (sc.hasNextLine()) {
            String line = sc.nextLine();
            String[] breakdown = line.split(" \\| ");
            String command = breakdown[0];

            try {
                TaskType taskType = TaskType.valueOf(command);

                switch (taskType) {
                case T:
                    tasks.add(new Todo(breakdown[2],
                            Boolean.parseBoolean(breakdown[1])));
                    break;
                case E:
                    tasks.add(new Event(breakdown[2],
                            Boolean.parseBoolean(breakdown[1]),
                            LocalDateTime.parse(breakdown[3])));
                    break;
                case D:
                    tasks.add(new Deadline(breakdown[2],
                            Boolean.parseBoolean(breakdown[1]),
                            LocalDateTime.parse(breakdown[3])));
                    break;
                }
            } catch (IllegalArgumentException e) {
                  throw new DukeException("INVALID TASK TYPE");
            }
        }
        return tasks;
         
```
Perhaps placing the try-catch block above the while-loop will be better to avoid _arrowhead_ style code?